Mono.Cecil.Cil/CodeWriter.cs

```diff
@@ -217,7 +217,7 @@ void WriteOperand (Instruction instruction)
 			case OperandType.ShortInlineBrTarget: {
 				var target = (Instruction) operand;
 				var offset = target != null ? GetTargetOffset (target) : body.code_size;
-				WriteSByte ((sbyte) (offset - (instruction.Offset + opcode.Size + 1)));
+				WriteSByte (checked ((sbyte) (offset - (instruction.Offset + opcode.Size + 1))));
```

Comment:
That might not be a good idea for the long forms of the Ldc instructions. Truncating `Ldc_i4_S 0x123` to `Ldc_i4_S 0x23` is almost certainly a bug and deserves throwing an exception, but what about unsigned integer constants like `Ldc_i4 0xffffffffu`? That's probably not a bug, because there is no separate Ldc instruction for unsigned integers, but `checked ((int) 0xffffffffu)` will throw. On the other hand, truncating significant bits off a `long` for `Ldc_i4` is just as bad as truncating significant digits off an `int` for `Ldc_i4`, not to mention silently casting a floating-point number to an integer, so there is a case for more thorough checking than I've added here.